keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/ResourceEvent.kt

```diff
@@ -367,18 +376,35 @@ data class ResourceCheckUnresolvable(
     )
 }
 
+enum class ResourceCheckErrorOrigin {
+  @JsonProperty("user") USER,
+  @JsonProperty("system") SYSTEM;
+
+  companion object {
+    fun fromException(exceptionType: Class<out SpinnakerException>) =
+      when {
+        UserException::class.java.isAssignableFrom(exceptionType) -> USER
+        SystemException::class.java.isAssignableFrom(exceptionType) -> SYSTEM
+        else -> throw SystemException("All keel exceptions should inherit from UserException or SystemException. This is a bug.")
```

Comment:
I think it would be better to log a message here, set this to "UNKNOWN", and then set up an alert to watch for the message. If I'm not mistaken, this code here means that If we miss something in a PR review or accidentally do it wrong, then the whole history view will error out.  

I strongly prefer logging messages when something is in a weird but not harmful state, because that means it's more likely a view will remain complete for users.

Reply:
Good point. Will change.